src/Proxy/LazyCollection.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace GraphAware\Neo4j\OGM\Proxy;
+
+use Doctrine\Common\Collections\AbstractLazyCollection;
+use GraphAware\Common\Type\Node;
+use GraphAware\Neo4j\OGM\Common\Collection;
+
+class LazyCollection extends AbstractLazyCollection
+{
+    protected $initalizer;
+
+    protected $node;
+
+    protected $object;
+
+    protected $initialized = false;
+
+    protected $initializing = false;
+
+    protected $added = [];
```

Comment:
Are these properties needed to be protected? We should consider make them private.